linux/runner/my_application.cc

```diff
@@ -100,31 +108,23 @@ static void my_application_dispose(GObject* object) {
 
 static void my_application_class_init(MyApplicationClass* klass) {
   G_APPLICATION_CLASS(klass)->activate = my_application_activate;
-  G_APPLICATION_CLASS(klass)->local_command_line =
-      my_application_local_command_line;
+  G_APPLICATION_CLASS(klass)->local_command_line = my_application_local_command_line;
+  G_APPLICATION_CLASS(klass)->startup = my_application_startup;
+  G_APPLICATION_CLASS(klass)->shutdown = my_application_shutdown;
   G_OBJECT_CLASS(klass)->dispose = my_application_dispose;
 }
 
 static void my_application_init(MyApplication* self) {}
 
-bool is_flatpak(void) {
-  if (getenv("container") || getenv("FLATPAK_ID") || getenv("FLATPAK")) {
-    /* flatpak */
-    return true;
-  }
-  return false; /* No container detected */
-}
-
 MyApplication* my_application_new() {
-  // gchar based alternate MY_APPLICATION_ID
-  const char* my_application_id = APPLICATION_ID;
-
-  if (is_flatpak()) {
-    my_application_id = "com.github.KRTirtho.Spotube";
-  }
-
-  return MY_APPLICATION(g_object_new(
-      my_application_get_type(), "application-id", my_application_id, "flags",
-      G_APPLICATION_HANDLES_COMMAND_LINE | G_APPLICATION_HANDLES_OPEN,
-      nullptr));
-}
\ No newline at end of file
+  // Set the program name to the application ID, which helps various systems
+  // like GTK and desktop environments map this running application to its
+  // corresponding .desktop file. This ensures better integration by allowing
+  // the application to be recognized beyond its binary name.
+  g_set_prgname(APPLICATION_ID);
+
+  return MY_APPLICATION(g_object_new(my_application_get_type(),
+                                     "application-id", APPLICATION_ID,
+                                     "flags", G_APPLICATION_NON_UNIQUE,
+                                     nullptr));
+}
```

Comment:
Please undo the changes in the file and move it back to where t was. This breaks the entire app on linux. Please undo it